### PR TITLE
[Merged by Bors] - Add try_* to add_slot_edge, add_node_edge

### PR DESCRIPTION
**Pull request by torsteingrindvik**
# Objective

`add_node_edge` and `add_slot_edge` are fallible methods, but are always used with `.unwrap()`.
`input_node` is often unwrapped as well.
This points to having an infallible behaviour as default, with an alternative fallible variant if needed.

Improves readability and ergonomics.

## Solution

- Change `add_node_edge` and `add_slot_edge` to panic on error.
- Change `input_node` to panic on `None`.
- Add `try_add_node_edge` and `try_add_slot_edge` in case fallible methods are needed.
- Add `get_input_node` to still be able to get an `Option`.
---

### Added

- `try_add_node_edge`
- `try_add_slot_edge`
- `get_input_node`

### Changed

- `add_node_edge` is now infallible (panics on error)
- `add_slot_edge` is now infallible (panics on error)
- `input_node` now panics on `None`

## Migration Guide

Remove `.unwrap()` from `add_node_edge` and `add_slot_edge`.
For cases where the error was handled, use `try_add_node_edge` and `try_add_slot_edge` instead.

Remove `.unwrap()` from `input_node`.
For cases where the option was handled, use `get_input_node` instead.

### CODE REVIEW

File: crates/bevy_render/src/render_graph/graph.rs [hunk 1]
```diff
@@ -297,7 +324,13 @@ impl RenderGraph {
 
     /// Adds the [`Edge::NodeEdge`] to the graph. This guarantees that the `output_node`
     /// is run before the `input_node`.
-    pub fn add_node_edge(
+    ///
+    /// Fails if any invalid [`NodeLabel`] is given.
+    ///
+    /// # See also
+    ///
+    /// - [`add_node_edge`](Self::add_node_edge) for a fallible version.
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/render_graph/graph.rs:332]:
```suggestion
    /// - [`add_node_edge`](Self::add_node_edge) for an infallible version.
```

**Review by alice-i-cecile — COMMENTED** [review]
On board with this direction: I like this as an API pattern a lot.

I'm noticing the same pattern `node_id().unwrap()` while reviewing. Should we change that too? Should it be in the same PR?

**Review by IceSentry — APPROVED** [review]
Oh wow, such a simple change but it makes the render graph a lot nicer!

**Review by alice-i-cecile — APPROVED** [review]

**Comment by torsteingrindvik** [comment]
@alice-i-cecile About `input_node()`, I agree.
I think `try_input_node()` would be misleading since the return is `Option` and not `Result` though.

I saw in rustlang that there is precedence for having a `checked_foo` for `Option<T>` when `foo` returns `T`.

So should I make the change for `input_node()` -> infallible, and add `input_node_checked()`?

**Comment by torsteingrindvik** [comment]
I'll use that then, it's unsurprising which is good.

I was a bit thrown off by this: https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

But I'm not sure what the justification is there and anyway I see Bevy uses a lot of `get_foo`.

I'll update the description.

**Comment by IceSentry** [comment]
Yeah, it's pretty much a bevy convention that when you want a fallible getter you add the `get_` otherwise you just name it without the prefix.

**Comment by torsteingrindvik** [comment]
Ok, that's good to know.
Description updated.

Hopefully CI passes, there was a strange error last run.

**Pull request closed without merging**